File: src/Services/Schedule/Schedule.Common/Models/Class.cs
using System;
using System.Collections.Generic;

namespace Schedule.Common.Models
{
    public class Class
    {
        public string Id { get; set; }
        public int Semester { get; set; }
        public int Year { get; set; }
        public string CourseId { get; set; }

        public string CourseName { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public List<ClassSchedule> ClassSchedules { get; set; }
    }
}

File: src/Web/Spa/src/app/models/class.ts
import { Course } from './course';

export class Class{
    id:string;
    semester:number;
    year:number;
    startDate:Date;
    endDate:Date;
    course:Course;
}

export class ShortClass{
    id:string;
    semester:number;
    year:number;
    courseID:string;
    courseName:string;
}